### PR TITLE
Buttsmithy

### LINKED ISSUE #10063: Buttsmithy
### Source name

Buttsmithy

### Source link

https://buttsmithy.com

### Language

en

### Other details

The website [https://buttsmithy.com](https://buttsmithy.com) contains multiple sources. (Though comics other than Alfie are found under the subdomain [https://incase.buttsmithy.com](https://incase.buttsmithy.com).)
This includes Alfie and other smaller comics from Incase. This source was requested two times already in the past (#3578, #2257) but both issues were closed due to being open for too long.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a title with source name.
- [X] I have checked that the extension does not already exist on the [website extensions list](https://tachiyomi.org/extensions/) or the app.
- [X] I have checked that the extension does not already exist by searching the [GitHub repository](https://github.com/tachiyomiorg/tachiyomi-extensions/) and verified it does not appear in the code base.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by THE-ORONCO**
This adds the source Buttsmithy as an extension and will close  #10063
The extension was both tested in Android Studio and on a real phone.

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [x] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

File: src/en/buttsmithy/src/eu/kanade/tachiyomi/extension/en/buttsmithy/Buttsmithy.kt [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+                    artist = inCase
+                    status = decideAlfieStatusFromTitle(chapTitle, mostRecentChapTitle)
+                    genre = "fantasy, NSFW"
+                    thumbnail_url = generateImageUrlWithText(alfieTitle)
+                }
+            } else manga
+        )
+    }
+
+    override fun mangaDetailsRequest(manga: SManga): Request = throw Exception("Not used")
```
Comment (ObserverOfTime) [inline src/en/buttsmithy/src/eu/kanade/tachiyomi/extension/en/buttsmithy/Buttsmithy.kt:282]: This is always used by the webview. (tachiyomiorg/tachiyomi#1755)

```suggestion
    override fun mangaDetailsRequest(manga: SManga) =
        if (manga.title.contains(alfieTitle)) GET(manga.url) else GET(baseUrl + manga.Url)
    
```

**Comment by CarlosEsco** [comment]
This git history is pretty fucked,  THere are tons of other changes besides buttsmithy.  You might want to just close the pr, and make a new branch from master and add the buttsmithy again and open the pr.

**Comment by ObserverOfTime** [comment]
A rebase and force push should fix it.

**Comment by THE-ORONCO** [comment]
> This git history is pretty fucked, THere are tons of other changes besides buttsmithy. You might want to just close the pr, and make a new branch from master and add the buttsmithy again and open the pr.

Yeah
I'll do that. I accidentally pulled while working on the extension. Sorry

**Pull request closed without merging**